db/migrate/20240326194731_migrate_wac_roles.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class MigrateWacRoles < ActiveRecord::Migration[7.1]
+  def create_user_role_for_team_member(team_member, group)
+    if team_member.leader?
+      status = RolesMetadataCouncils.statuses[:leader]
+    elsif team_member.senior_member?
+      status = RolesMetadataCouncils.statuses[:senior_member]
+    else
+      status = RolesMetadataCouncils.statuses[:member]
+    end
+    metadata = RolesMetadataCouncils.create!(status: status)
+    UserRole.create!(
+      user_id: team_member.user_id,
+      group_id: group.id,
+      start_date: team_member.start_date,
+      end_date: team_member.end_date,
+      metadata: metadata,
+    )
+  end
+
+  def change
+    UserGroup.councils.each do |council|
+      Team.find_by(friendly_id: council.metadata.friendly_id).team_members.each do |member|
```

Comment:
Attention: `Team` has a "default scope" defined. It means that every query you do will have `WHERE hidden = 0` attached, **even when you do stuff like `Team.all`**.

In this case, it will be fine because we don't have a "hidden council". But when you migrate T/C memberships in the future, you probably also want to migrate WCT China and WST Admin, in which case you need to do `Team.with_hidden.find_by`

Reply:
Got it, thanks.